packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -462,18 +477,61 @@ export class BuildService {
     );
     const { resourceInfo } = dSGResourceData;
 
-    const resourceRepository = await this.resourceService.gitRepository(
-      build.resourceId
-    );
+    const project = await this.prisma.project.findUnique({
+      where: {
+        id: resource.projectId,
+      },
+    });
 
-    if (!resourceRepository) {
-      return;
-    }
+    let gitSettings: CreatePullRequestGitSettings = null;
+    let kafkaEventKey: string = null;
 
-    const gitOrganization =
-      await this.resourceService.gitOrganizationByResource({
-        where: { id: resource.id },
-      });
+    if (project.useDemoRepo) {
+      const organizationName = this.configService.get<string>(
+        Env.GITHUB_DEMO_REPO_ORGANIZATION_NAME
+      );
+
+      const installationId = this.configService.get<string>(
+        Env.GITHUB_DEMO_REPO_INSTALLATION_ID
+      );
+
+      gitSettings = {
+        gitOrganizationName: organizationName,
+        gitRepositoryName: project.demoRepoName,
+        repositoryGroupName: "",
+        gitProvider: EnumGitProvider.Github,
+        gitProviderProperties: {
+          installationId: installationId,
+        },
+      };
+
+      kafkaEventKey = project.demoRepoName;
+    } else {
+      const resourceRepository = await this.resourceService.gitRepository(
+        build.resourceId
+      );
+
+      kafkaEventKey = resourceRepository.id;
```

Comment:
@overbit I need you to look at this change and confirm that it is OK to set the kafkaEventKey this way.
Originally it was always "resourceRepository.id;"
now, since we may not have an resourceRepository object (when using demo repo) - so I used the demoRepoName (which is the actual name of the repository in the demo organization)

is that ok?